<h3>Computational Physics 2</h3>
<p>Reymond R. Mesuga</p>

<h3>Monte Carlo Integration</h3>

In [1]:
%matplotlib inline
from scipy import random
import numpy as np
import matplotlib.pyplot as plt

<h3>Problem 1</h3>

<p>When N = 100</p>

In [2]:
a = 1
b = 2
N = 100
xrand = np.zeros(N)

for i in range(len(xrand)):
    xrand[i] = random.uniform(a,b)
def func(x):
    return np.exp(x)
integral = 0.0

for i in range(N):
    integral += func(xrand[i])

answer = (b-a)/float(N)*integral
answer

4.5312401312764745

<p>When N = 500</p>

In [3]:
N = 500
xrand = np.zeros(N)

for i in range(len(xrand)):
    xrand[i] = random.uniform(a,b)
def func(x):
    return np.exp(x)
integral = 0.0

for i in range(N):
    integral += func(xrand[i])

answer = (b-a)/float(N)*integral
answer

4.741867063478121

<p>When N = 1000</p>

In [4]:
N = 1000
xrand = np.zeros(N)

for i in range(len(xrand)):
    xrand[i] = random.uniform(a,b)
def func(x):
    return np.exp(x)
integral = 0.0

for i in range(N):
    integral += func(xrand[i])

answer = (b-a)/float(N)*integral
answer

4.7143157103642315

<h3>Problem 2</h3>

<p>When N = 100</p>

In [12]:
x0 = 1
x1 = 2
y0 = 1
y1 = 2
n = 100
x = np.random.uniform(x0, x1, n)
y = np.random.uniform(y0, y1, n)
def f(x,y):
    return np.exp(x*y)
def g(x,y):
    return(np.exp(x*y) if (1 <= x <= 2 and 1 <= y <= 2) else -np.exp(x*y))

f_mean = 0
num_inside = 0
for i in range(len(x)):
    for j in range(len(y)):
        if g(x[i], y[j]) >= 0:
            num_inside += 1
            f_mean += f(x[i], y[j])
f_mean = f_mean/float(num_inside)
area = num_inside/float(n**2)*(x1 - x0)*(y1 - y0)
area*f_mean

11.611004394969791

<p>When N = 500</p>

In [7]:
x0 = 1
x1 = 2
y0 = 1
y1 = 2
n = 500
x = np.random.uniform(x0, x1, n)
y = np.random.uniform(y0, y1, n)
def f(x,y):
    return np.exp(x*y)
def g(x,y):
    return(np.exp(x*y) if (1 <= x <= 2 and 1 <= y <= 2) else -np.exp(x*y))

f_mean = 0
num_inside = 0
for i in range(len(x)):
    for j in range(len(y)):
        if g(x[i], y[j]) >= 0:
            num_inside += 1
            f_mean += f(x[i], y[j])
f_mean = f_mean/float(num_inside)
area = num_inside/float(n**2)*(x1 - x0)*(y1 - y0)
area*f_mean

11.79730697636514

<p>When N = 1000</p>

In [15]:
x0 = 1
x1 = 2
y0 = 1
y1 = 2
n = 1000
x = np.random.uniform(x0, x1, n)
y = np.random.uniform(y0, y1, n)
def f(x,y):
    return np.exp(x*y)
def g(x,y):
    return(np.exp(x*y) if (1 <= x <= 2 and 1 <= y <= 2) else -np.exp(x*y))

f_mean = 0
num_inside = 0
for i in range(len(x)):
    for j in range(len(y)):
        if g(x[i], y[j]) >= 0:
            num_inside += 1
            f_mean += f(x[i], y[j])
f_mean = f_mean/float(num_inside)
area = num_inside/float(n**2)*(x1 - x0)*(y1 - y0)
area*f_mean

11.765375410696716

<h3>Conclusion</h3>
<p>In problem 1, the theoretical value matches the approximated value solved by monte carlo integration method. At N=100, the random value range from 4 to 5 with corresponding decimals. At N=500 and N=1000, the range of random values sticks within approximately 4.7 with corresponding decimals. In problem 2, at N=100, the random value given by monte carlo method gives 11 to 12 with corresponding decimals. At N=500 and N=1000, the random value sticks within approxiamtely 11.7 with corresponding decimals. Overall, it is observed that random value solve by monte carlo integration become consistent as the value of N increases.</p>